src/utils/embedded/embedded.utils.ts

```diff
@@ -72,21 +87,20 @@ const {
 // type TRPCClient = ReturnType<typeof createTRPCProxyClient<AppRouter>>;
 // const trpcVanilla = client as TRPCClient;
 
-const supabase = IS_EMBEDDED_APP
-  ? createSupabaseClient(
+function getSupabaseClient() {
```

Comment:
Now that you are here, I'd like to request one more change: Make this function async and provide a custom storage instance for supasbase that uses unpartitioned state (you can see a commented out draft for that in `src/iframe/plasmo-storage/plasmo-storage.mock.ts`).